src/main/java/com/laba/solvd/task3/pages/HomePage.java

```diff
@@ -0,0 +1,83 @@
+package com.laba.solvd.task3.pages;
+
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import com.zebrunner.carina.webdriver.decorator.ExtendedWebElement;
+import com.zebrunner.carina.webdriver.gui.AbstractPage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+import org.testng.log4testng.Logger;
+
+public class HomePage extends AbstractPage implements IMobileUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(HomePage.class);
+
+    @FindBy(xpath = "//*[@resource-id='collectionItemCatalog']")
+    private ExtendedWebElement catalogButton;
+
+    @FindBy(xpath = "//*[@resource-id='catalogItemText' and @text='%s']")
+    private ExtendedWebElement catalogItemName;
+
+    @FindBy(xpath = "//*[@resource-id='add_productTab']")
+    private ExtendedWebElement addNewProductButton;
+
+    @FindBy(xpath = "//android.widget.HorizontalScrollView/*[%d]")
+    private ExtendedWebElement vipProductCard;
+
+    @FindBy(xpath = "//*[@resource-id='profileTab']")
+    private ExtendedWebElement profileButton;
+
+    @FindBy(xpath = "//*[contains(@resource-id,'fragmentContainer')]/androidx.compose.ui.platform.ComposeView/android.view.View/android.view.View/android.view.View[1]/android.view.View[4]")
+    private ExtendedWebElement searchButton;
+
+    public HomePage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public boolean isPageOpened() {
+        return catalogButton.isPresent();
+    }
+
+    public CategoryPage openCatalogCategory(String categoryName) {
+        LOGGER.info(catalogItemName.format(categoryName).isPresent(3));
+        if (!catalogItemName.format(categoryName).isPresent(3)) {
+            ExtendedWebElement lastCatalogItem = findExtendedWebElement(By.xpath("(//*[@resource-id='catalogBoxItem'])[2]/*"));
```

Comment:
you don't need to create variable lastCatalogItem here, you can create this locator as usually as parameter of your class and just use necessary methods